overlord/state/taskrunner.go

```diff
@@ -73,13 +85,15 @@ func (r *TaskRunner) run(fn HandlerFunc, task *Task) {
 
 		r.state.Lock()
 		defer r.state.Unlock()
+		halted := task.HaltTasks()
 		if err == nil {
 			task.SetStatus(DoneStatus)
 		} else {
-			task.SetStatus(ErrorStatus)
+			taskFail(task)
+		}
+		if len(halted) > 0 {
+			r.state.EnsureBefore(0)
```

Comment:
well there's also the done case, it will trigger the new goroutines (and other booking that is best done in Ensure)